Source/PTEConsoleLogger.m

```diff
@@ -99,14 +99,20 @@ - (instancetype)init
 - (void)logMessage:(DDLogMessage *)logMessage
 {
     // The method is called from the logger queue
+    __weak typeof(self) weakSelf = self;
     dispatch_async(_consoleQueue, ^
     {
+        if (weakSelf == nil) {
+            return;
+        }
+
+        __strong typeof(weakSelf) strongSelf = weakSelf;
         // Add new message to buffer
-        [_newMessagesBuffer insertObject:logMessage
+        [strongSelf->_newMessagesBuffer insertObject:logMessage
```

Comment:
It is a `dispatch_async`, everything will be released once it is executed. No real need to use weak references IMO.